src/shutdown/shutdown.c

```diff
@@ -574,6 +574,13 @@ int main(int argc, char *argv[]) {
                           need_dm_detach ? " DM devices," : "",
                           need_md_detach ? " MD devices," : "");
 
+        /* This is primarily useful when running systemd in a VM, as it provides the user running the VM with
+         * a mechanism to pick up systemd's exit status in the VM. Note that we execute this before the final
+         * sync(), since otherwise we might shut down the VM before the AF_VSOCK buffers have been flushed.
+         * While sync() doesn't guarantee our message will arrive, it does seem to take long enough to make
+         * sure the message arrives on the listening socket. */
+        (void) sd_notifyf(0, "EXIT_STATUS=%i", arg_exit_code);
```

Comment:
hmm, why not do this even earlier? is there any reason not to do this as one of the first things when we enter main()?

Reply:
The host will receive the message earlier. That's the only difference really. Moved it up so we do this right after parsing all the arguments.